spynnaker/pyNN/models/defaults.py

```diff
@@ -182,3 +186,103 @@ def defaults(cls: type) -> type:
         cls.default_parameters = get_map_from_init(init, include=params)
         cls.default_initial_values = get_map_from_init(init, include=svars)
     return cls
+
+
+class AbstractProvidesDefaults(object):
+    """
+    Provides the default_parameters and default_initial_values properties
+
+    These will be filled in based on the @default_parameters and
+    @default_initial_values decorators with values read from the init.
+    """
+
+    @classmethod
+    def __fill_in_defaults(cls):
+        """
+        Fills in default_parameters and default_initial_values attributes
+        """
+        # get the init method
+        init = getattr(cls, "__init__")
+        # Find the real method as there may be decorators
+        while hasattr(init, "_method"):
+            init = getattr(init, "_method")
```

Comment:
Does this terminate?  How does the attribute get removed?

Reply:
I assume it does terminate as it is the same code as before... sometimes my own code confuses me!

Reply:
Oh I see now, it is essentially recusing the init until it finds the real init